### PR TITLE
Override 'DOMParser.parseFromString' function (close #1133)

### LINKED ISSUE #1133: We should override the 'DOMParser.DOMParser.parseFromString'
https://developer.mozilla.org/en-US/docs/Web/API/DOMParser

### PR DESCRIPTION
**Pull request by LavrovArtem**
/cc @churkin @miherlosev

### CODE REVIEW

File: src/client/sandbox/node/window.js [hunk 1]
```diff
@@ -326,6 +326,15 @@ export default class WindowSandbox extends SandboxBase {
             };
         }
 
+        if (window.DOMParser) {
+            window.DOMParser.prototype.parseFromString = function (...args) {
+                if (args.length > 1 && typeof args[0] === 'string')
```
Comment (miherlosev) [inline src/client/sandbox/node/window.js]: `args.length` is an uncessary check. Drop this.

Also, as I understand we should proxy a string only for the ` text/html` type.

For another supported types:
    "text/xml",
    "application/xml",
    "application/xhtml+xml",
    "image/svg+xml"
 we should not process string.

We have an issue for XHTML  support - https://github.com/DevExpress/testcafe-hammerhead/issues/350

@churkin

Comment (LavrovArtem) [inline src/client/sandbox/node/window.js]: I'm disagree with
> args.length is an uncessary check. Drop this.

The `parseFromString` function throw an error `Failed to execute 'parseFromString' on 'DOMParser': 2 arguments required, but only 1 present.` and we don't need to process the string.

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

File: src/client/sandbox/node/window.js [hunk 2]
```diff
@@ -326,6 +326,15 @@ export default class WindowSandbox extends SandboxBase {
             };
         }
 
+        if (window.DOMParser) {
+            window.DOMParser.prototype.parseFromString = function (...args) {
```
Comment (miherlosev) [inline src/client/sandbox/node/window.js]: Why do you override the `parseFromString` from `DOMParser.prototype`?

File: src/client/sandbox/native-methods.js [hunk 3]
```diff
@@ -173,6 +173,10 @@ class NativeMethods {
         // Object
         this.objectToString = win.Object.prototype.toString;
 
+        // DOMParser
+        if (win.DOMParser)
+            this.DOMParserParseFromString = win.DOMParser.prototype.parseFromString;
```
Comment (miherlosev) [inline src/client/sandbox/native-methods.js]: Why do you store value from prototype?

File: src/client/sandbox/node/window.js (hunk 2 above)
Comment (LavrovArtem) [inline src/client/sandbox/node/window.js]: What do you meen?

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

File: test/client/fixtures/sandbox/node/methods-test.js [hunk 4]
```diff
@@ -320,6 +320,28 @@ if (!browserUtils.isFirefox) {
     });
 }
 
+if (window.DOMParser) {
+    test('DOMParser.parseFromString', function () {
+        var htmlStr        = '<a href="/path">Link</a>';
+        var parsedDocument = new DOMParser().parseFromString(htmlStr, 'text/html');
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/node/methods-test.js]: Small style issue.

Rewrite as
```js
var domParser           = new DOMParser();
var parsedDocument = domParser.parseFromString(htmlStr, 'text/html');
```
and remove `new DOMParser()`  below

**Comment by LavrovArtem** [comment]
FPR

**Comment by LavrovArtem** [comment]
ping @churkin

File: src/client/sandbox/node/window.js [hunk 5]
```diff
@@ -326,6 +326,16 @@ export default class WindowSandbox extends SandboxBase {
             };
         }
 
+        // NOTE: DOMParser supports an HTML parsing for IE10 and later
+        if (window.DOMParser && !isIE9) {
+            window.DOMParser.prototype.parseFromString = function (...args) {
+                if (args.length > 1 && typeof args[0] === 'string' && args[1] === 'text/html')
```
Comment (churkin) [inline src/client/sandbox/node/window.js]: `args.length` is extra check

Comment (churkin) [inline src/client/sandbox/node/window.js]: Sorry, I was wrong

**Pull request merged**